pkg/updatestatus/nodeinformer.go

```diff
@@ -152,7 +161,23 @@ func (c *nodeInformerController) sync(ctx context.Context, syncCtx factory.SyncC
 		}
 		return nil
 	case machineConfigPoolKindName:
-		return c.reconcileAllNodes(syncCtx.Queue())
+		machineConfigPool, err := c.machineConfigPools.Get(name)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+		if kerrors.IsNotFound(err) {
+			// The pool was deleted
+			if changed := c.machineConfigPoolSelectorCache.forget(name); changed {
+				klog.V(2).Infof("Reconciling all nodes as machine config pool %q is deleted", name)
+				return c.reconcileAllNodes(syncCtx.Queue())
+			}
+			return nil
+		}
+		if changed := c.machineConfigPoolSelectorCache.ingest(machineConfigPool); changed {
+			klog.V(2).Infof("Reconciling all nodes as machine config pool %q is refreshed", name)
+			return c.reconcileAllNodes(syncCtx.Queue())
+		}
```

Comment:
consider similar structural changes I suggested for MC one

Reply:
Done.

https://github.com/openshift/cluster-version-operator/pull/1164/commits/a81ec69e31f504ce10a44650cfa5c4d4260fd2cd#diff-c5a41cb83200e13544e0509a5e77c180e16a1d70d7411a79cac2bc107e89950fR295